src/Tools/dotnet-monitor/HostBuilder/HostBuilderHelper.cs

```diff
@@ -87,6 +89,18 @@ public static IHostBuilder CreateHostBuilder(HostBuilderSettings settings)
                         }
                     }
                 })
+                .ConfigureAppConfiguration((HostBuilderContext context, IConfigurationBuilder builder) =>
+                {
+                    // Save and clear URLs so ASP.NET app configuration cannot see them
+                    appUrls = context.Configuration[WebHostDefaults.ServerUrlsKey];
```

Comment:
Might be able to do better here by checking which configuration source provides the value, save the source and the value, and clear the value on the source. Then restore the value on the source after configuring ASP.NET so that the perceived configuration provider is not changed. The current implementation should not affect product behavior, but it does change the configuration source of the value as seen by `config show --show-sources`.